src/components/run-button.js

```diff
@@ -5,104 +5,106 @@
             // a computation may become unavailable when developer mode is disabled, then switch on first one
-            setSelectedIndex(0);
+            setSelectedRunnable(activeRunnables[0]);
         }
-    }, [enableDeveloperMode, setSelectedIndex]);
+    }, [activeRunnables, selectedRunnable, setSelectedRunnable]);
 
     const getRunningStatus = useCallback(() => {
-        return props.getStatus(getRunnable());
-    }, [getRunnable, props]);
-
-    let buttonDisabled =
-        props.disabled ||
-        (selectedIndex === 0 && getRunningStatus() !== RunningStatus.IDLE) ||
-        (selectedIndex === 1 && isRunning()) ||
-        (selectedIndex === 4 /* Dynamic simulation button is selected */ &&
-            props.getStatus(props.runnables[0]) !==
-                RunningStatus.SUCCEED); /* Load flow button's status must SUCCEED */
+        return getStatus(selectedRunnable);
+    }, [selectedRunnable, getStatus]);
 
-    function handleActionOnRunnable() {
-        props.actionOnRunnable.action(getRunnable());
+    function isButtonDisable() {
+        if (selectedRunnable === ComputingType.LOADFLOW) {
+            // We run once loadflow analysis, as it will always return the same result for one hypothesis
+            return getRunningStatus() !== RunningStatus.IDLE;
+        } else if (selectedRunnable === ComputingType.SECURITY_ANALYSIS) {
+            // We can run only 1 analysis at a time
+            return getRunningStatus() === RunningStatus.RUNNING;
+        } else if (selectedRunnable === ComputingType.DYNAMIC_SIMULATION) {
+            // Load flow button's status must be "SUCCEED"
+            return getStatus(ComputingType.LOADFLOW) !== RunningStatus.SUCCEED;
+        } else {
+            return false;
+        }
     }
 
     return (
         <SplitButton
-            options={getOptions(
-                getRunningStatus(),
-                props.runnables,
-                props.actionOnRunnable
-            )}
-            selectedIndex={selectedIndex}
-            onSelectionChange={(index) => setSelectedIndex(index)}
-            onClick={handleClick}
-            runningStatus={getRunningStatus()}
-            buttonDisabled={buttonDisabled}
-            selectionDisabled={props.disabled}
-            text={
-                props.getText
-                    ? props.getText(getRunnable(), getRunningStatus())
-                    : ''
+            options={getOptions()}
+            selectedIndex={activeRunnables.indexOf(selectedRunnable)}
+            onSelectionChange={(index) =>
+                setSelectedRunnable(activeRunnables[index])
             }
-            actionOnRunnable={handleActionOnRunnable}
+            onClick={runnables[selectedRunnable].startComputation}
+            runningStatus={getRunningStatus()}
+            buttonDisabled={disabled || isButtonDisable()}
+            selectionDisabled={disabled}
+            text={runnablesText[selectedRunnable] || ''}
```

Comment:
The label for the selected runnable isn't showing up, I reckon it is caused by trying to access `runnablesText` with `selectedRunnable` although `runnablesText` is a flat array

Reply:
yes that's it